File: core/src/com/mygdx/game/deckengine/cards/DefendCard.java
package com.mygdx.game.deckengine.cards;

import com.badlogic.gdx.graphics.Texture;
import com.mygdx.game.character.CharacterStats;
import com.mygdx.game.character.CharacterType;

public class DefendCard extends Card {

    public DefendCard(Texture img) {
        super(img);
    }

    @Override
    public boolean CanApplyEffects(CharacterStats stats) {
        return (stats.getCharacterType() == CharacterType.PLAYER);
    }

    @Override
    public void ApplyEffects(CharacterStats stats) {
        stats.AdjustBlock(5);
    }
}

File: deck/src/classicthunder/card/impl/DefendCard.kt
package classicthunder.card.impl

import classicthunder.card.Card
import classicthunder.character.CharacterStats
import classicthunder.character.CharacterType

class DefendCard(parent: Card?) : Card(parent) {

    private var block: Int

    init {
        block = 5
    }

    override fun getEnergyCost(): Int {
        return 1
    }

    override fun canPlay(stats: CharacterStats): Boolean {
        return stats.characterType == CharacterType.PLAYER
    }

    override fun play(stats: CharacterStats) {
        stats.adjustBlock(block)
    }

    override fun copy(parent: Card): DefendCard {
        val copy = DefendCard(parent)
        copy.block = this.block
        return copy
    }
}